spec/features/generic_announcements_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.feature 'Generic Announcements' do
+  subject { page }
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    let!(:user) { create(:user) }
+
+    context 'As a User' do
+      before(:each) do
+        login_as(user, scope: :user)
+        instance.announcements.clear
+        SystemAnnouncement.destroy_all
+      end
+
+      scenario 'I should not see any announcements if there are none' do
+        visit announcements_path
+        it { is_expected.not_to have_selector('div.instance-announcement') }
+        it { is_expected.not_to have_selector('div.system-announcement') }
+      end
+
+      scenario 'I should see instance announcements' do
+        create(:instance_announcement, instance: instance, creator: user, updater: user)
+        visit announcements_path
+        it { is_expected.to have_selector('div.instance-announcement') }
+        it { is_expected.to render_template('index') }
+      end
+
+      scenario 'I should see system announcements' do
+        create(:system_announcement, creator: user, updater: user)
+        visit announcements_path
+        it { is_expected.to have_selector('div.system-announcement') }
+      end
+
+      scenario 'I should see both types of announcements in the same view' do
+        create_list(:system_announcement, 5, creator: user, updater: user)
+        create_list(:instance_announcement, 5, instance: instance, creator: user, updater: user)
+        visit announcements_path
+
+        5.times do |i|
```

Comment:
you should iterate over the list of created announcements more than anything else.

The ID of the announcement div should be generated based on the database record ID, not the index of the page...